src/proctored-exam-settings/ProctoredExamSettings.test.jsx

```diff
@@ -123,3 +124,95 @@ describe('ProctoredExamSettings connection states tests', () => {
+    data: {
+      proctored_exam_settings: {
+        enable_proctored_exams: true,
+        allow_proctoring_opt_out: false,
+        proctoring_provider: 'mockproc',
+        proctoring_escalation_email: 'test@example.com',
+        create_zendesk_tickets: true,
+      },
+      available_proctoring_providers: ['software_secure', 'proctortrack', 'mockproc'],
+    },
+  };
+
+  function mockAPI(getData, postResult) {
+    const mockClientGet = jest.fn(async () => (getData));
+    const mockClientPost = postResult ? jest.fn(async () => (postResult)) : jest.fn(async () => { throw new Error(); });
+    auth.getAuthenticatedHttpClient = jest.fn(() => ({
+      get: mockClientGet,
+      post: mockClientPost,
+    }));
+    auth.getAuthenticatedUser = jest.fn(() => ({ userId: 3 }));
+    return { mockClientGet, mockClientPost };
+  }
+
+  it('Makes API call successfully', async () => {
+    const mockedFunctions = mockAPI(mockGetData, { data: 'success' });
+    await act(async () => render(<ProctoredExamSettings {...defaultProps} />));
+    // Make a change to the provider to proctortrack and set the email
+    const selectElement = screen.getByDisplayValue('mockproc');
+    await act(async () => {
+      fireEvent.change(selectElement, { target: { value: 'proctortrack' } });
+    });
+    const escalationEmail = screen.getByTestId('escalationEmail');
+    expect(escalationEmail.value).toEqual('test@example.com');
+    await act(async () => {
+      fireEvent.change(escalationEmail, { target: { value: 'proctortrack@example.com' } });
+    });
+    expect(escalationEmail.value).toEqual('proctortrack@example.com');
+    const submitButton = screen.getByTestId('submissionButton');
+    await act(async () => {
+      fireEvent.click(submitButton, { target: { value: 'proctortrack@example.com' } });
+    });
+    expect(mockedFunctions.mockClientPost).toHaveBeenCalled();
+    expect(mockedFunctions.mockClientPost).toHaveBeenCalledWith(
+      StudioApiService.getServiceUrl(defaultProps.courseId),
+      {
+        proctored_exam_settings: {
+          enable_proctored_exams: true,
+          allow_proctoring_opt_out: false,
+          proctoring_provider: 'proctortrack',
+          proctoring_escalation_email: 'proctortrack@example.com',
+          create_zendesk_tickets: false,
+        },
+      },
+    );
+    const errorAlert = screen.getByTestId('saveSuccess');
+    expect(errorAlert.textContent).toEqual(
+      expect.stringContaining('Proctored exam settings saved successfully.'),
+    );
+  });
+
+  it('Makes API call generated error', async () => {
+    const mockedFunctions = mockAPI(mockGetData, false);
+    await act(async () => render(<ProctoredExamSettings {...defaultProps} />));
+    // Make a change to the provider to proctortrack and set the email
+    const selectElement = screen.getByDisplayValue('mockproc');
+    await act(async () => {
+      fireEvent.change(selectElement, { target: { value: 'proctortrack' } });
```

Comment:
Is there a reason that the input changes are being tested? I feel like the purpose of this test is to check for the alert, which seems unrelated to input changes.

Reply:
That's a good point.

Reply:
I just want to test a change on the UI and make sure the POST data captures that change. That's why I am firing this event.

Reply:
I don't think the POST data is being checked in this test, though. I agree that a change on the UI makes sense for the test before this.

Reply:
@alangsto You are right. I have removed these.